demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/EventMapBehaviour.java

```diff
@@ -0,0 +1,470 @@
+    private FragmentManager supportFragmentManager;
+    private LocationRetriever locationRetriever;
+
+
+    private AbstractAddEventDialog addEventDialog;
+    private AbstractRemoveEventDialog removeEventDialog;
+    protected BottomSheetBehaviour bottomSheetBehaviour;
+    private View goToNavigatorView;
+    private GoToGoogleMapsNavigator goToGoogleMapsNavigator;
+
+    private List<OnEventCreatedListener> onEventCreatedListeners;
+    private List<OnEventRemovedListener> onEventRemovedListeners;
+    private List<OnEventTriggeredListener> onEventTriggeredListeners;
+    private OnMapInitializedListener onMapInitializedListener;
+
+    private Marker currentFocusMarker;
+
+    private static final String CREATE_EVENT_DIALOG_TAG = "CREATE_EVENT_DIALOG_TAG";
+    private static final String REMOVE_EVENT_DIALOG_TAG = "REMOVE_EVENT_DIALOG_TAG";
+
+    /**
+     * Constructor that creates a fully operative EventMapBehaviour object
+     *
+     * To link this object to an existing map it's necessary to call mapFragment.getMapAsync(),
+     * passing an instance of this class, this will trigger the creation of the map view
+     *
+     * Note that operations accessing the map can be done once that has been built (obv) and it has
+     * been initialized, set an OnMapInitializedListener to get notified of the complete initialization
+     * of the map
+     */
+    public EventMapBehaviour() {
+
+    }
+
+    /**
+     * @param supportFragmentManager Needed to interact with the user through dialogs
+     */
+    @Override
+    public void setSupportFragmentManager(@NonNull FragmentManager supportFragmentManager) {
+        this.supportFragmentManager = supportFragmentManager;
+    }
+
+    /**
+     * @param locationRetriever An instance of an object that implements locationRetriever to get the gps position
+     */
+    @Override
+    public void setLocationRetriever(@NonNull LocationRetriever locationRetriever) {
+        this.locationRetriever = locationRetriever;
+        this.locationRetriever.setOnLocationAvailableListener(this);
+    }
+
+    /**
+     * @param dialog An object of a class that extends AbstractAddEventDialog
+     */
+    @Override
+    public void setAddEventDialog(@NonNull AbstractAddEventDialog dialog) {
+        this.addEventDialog = dialog;
+        this.addEventDialog.setResultListener(this);
+    }
+
+    /**
+     * @param dialog An object of a class that extends AbstractRemoveEventDialog
+     */
+    @Override
+    public void setRemoveEventDialog(@NonNull AbstractRemoveEventDialog dialog) {
+        this.removeEventDialog = dialog;
+        this.removeEventDialog.setRemoveEventListener(this);
+    }
+
+    /**
+     * Note that for successfully open a google Maps instance a GoToGoogleMapsNavigator object must be set
+     * This Method will set the view as INVISIBLE, it will turn visible when the user clicks on a Marker,
+     * and turn invisible again when another part of the map will be clicked

```

Comment:
```suggestion
     * Note that for successfully open a google Maps instance a GoToGoogleMapsNavigator {@link #goToNavigatorView} object must be set
     *
     * This Method will set the view as INVISIBLE, it will turn visible when the user clicks on a Marker,
     * and turn invisible again when another part of the map will be clicked
```

Reply:
`goToNavigatorView` is the View object that triggers the opening of the google maps navigation, it should be {@link GoToGoogleMapsNavigator } as you pointed out